Attempt at addressing: Announce GPS accuracy #103

ajiteshbankulaa opened the pull request:
Summary

In reference to [soundscape-community/soundscape#103](https://github.com/soundscape-community/soundscape/issues/103), I’ve implemented a new behavior that improves user awareness of GPS accuracy during startup and wake events.

Implementation Details

Startup/Wake Behavior:
On app startup or wake, the system now checks the GPS accuracy.

If the accuracy is better than the defined threshold (±10 meters), the app will announce that GPS accuracy is good.

If the accuracy is poor, the app will inform the user and suggest moving around to improve signal quality.

Once the accuracy improves beyond the threshold, the app announces that the GPS accuracy has improved.

User Preference Setting:
A new setting has been added to allow users to enable or disable GPS accuracy announcements.

By default, this setting is on false to prevent extra notification anncouncments from the GPS Accuracy feature for those who dont want.




But for users who prefer the GPS Acc. updates they can though enable this option to hear accuracy feedback on startup or wake.

<img width="370" height="632" alt="GPS Accuracy Setting" src="https://github.com/user-attachments/assets/7c8112fb-0295-418d-948e-80953839701b" />

Manual Accuracy Check Button:
A new button allows users to manually check the current GPS accuracy at any time.

This gives users more control while minimizing potential over-announcements.

<img width="349" height="743" alt="Manual Accuracy Button" src="https://github.com/user-attachments/assets/aca1079c-811c-4b65-ab32-6894fb2500f2" />
Notes

Ovreall this aims to adress issue #103 by improve accessibility for users who want GPS feedback, while being flexibile for those who prefer a quieter experience and dont care for it being alwasy on startup.

Any feedback or suggestions for improvement would be greatly appreciated.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * GPS accuracy announcements: spoken accuracy at startup and on improvement, plus a new toggle in Settings → Callouts and updated settings UI.

* **Documentation**
  * Added/expanded localizations for GPS announcements, FAQ two‑finger double‑tap tip, mail providers and place/landmark categories across multiple languages.

* **Bug Fixes**
  * Minor localization formatting cleanups; one locale contains a duplicated key that should be resolved.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ajiteshbankulaa commented:
cool thanks for the feedback, I'll try and fix all these issues soon

ajiteshbankulaa commented:
I like the idea here: "Poor GPS accuracy, exceeded 20 meter limit, move for better accuracy.". It makes sense I dought people will care for the specific number. So I did change that. I also added the GPS accuracy is ok part also. 

The button is removed now and also fixed all the labeling and whatnot issues form before I believe. 

as for the all cases annouce as:
GPS accuracy: %@ meters

Im personaly open to anything based on what everyone thinks would be better. Another thing is I could also put another toggle or something in settings so the user  can decide which way they would  like it. It should be pretty simple to do so, but I don't know if it would feel janky then. But i'm overall open to whatever.

JJGatchalian commented:
The "ok" accuracy message is still missing the comma found in the "good" accuracy message. That should look more like this: "GPS accuracy is ok, about %d meters." I think you need to change the ranges for "good," "ok," and "poor" to match the existing explanation in the "Troubleshooting" section. "Less than +-10 meters is good. Less than +-20 meters is ok. Anything else is poor". That means that the poor message should be above 20 meters or above 60 feet. The OK range should be between 10 and 20 meters, between 30 and 60 feet.

JJGatchalian commented:
On second thought, the simple version of just announcing the GPS accuracy without the good, fair, poor scale isn't as beginner friendly. Voice Vista already announces accuracy like that, and we want Soundscape to be more beginner friendly than Voice Vista, but without sounding patronizing for experienced users. We're continuing to follow the direction from when Soundscape used to be a Microsoft App, and I think our current version looks like a Microsoft feature in spirit. Don't worry about adding a button to change the announcement style. We'll have too many buttons to overcomplicate the UI.

JJGatchalian commented:
Reading the code back, it looks like it may need some serious cleaning. I don't understand the logic for the feet vs meters message the way this current pull request is written. I checked Weblate, and currently there is no string that just says "feet" or "meters" all by itself. Are you trying to say that the word "feet" is part of the "%@" value or not? If it is part of the value, we may need to add new strings to the localization that just say "meters" or "feet" or maybe "%@ meters" and "%@ feet" so that they could be translated. If not, the word "meters" or "feet" should be part of the message itself and not part of the value "%@", and we still need separate messages for the units of measure.

RDMurray commented on apps/ios/GuideDogs/Assets/Localization/de-DE.lproj/Localizable.strings:
```diff
@@ -4388,3 +4388,4 @@
 "mail.default" = "Standard";
 "mail.gmail" = "Gmail";
 "mail.msoutlook" = "Microsoft Outlook";
+
```
same as above

RDMurray commented on apps/ios/GuideDogs/Assets/Localization/en-US.lproj/Localizable.strings:
Same as above, change unit specific strings to generic ones.

JJGatchalian commented:
So you're saying that we should still use language formatter, but the distance should be capped at 20 meters or 60 feet for poor accuracy. I'm just making sure that it lines up with the Troubleshooting messages "+-10 meters is a good distance, +- 20 meters is ok, anything else is poor." I don't want Soundscape to accidentaly say something like "GPS accuracy is poor and above 35 meters..."

ajiteshbankulaa commented:
Hello, 
Based on what I understood, the issues were:

1) Remove unit-specific strings
So I went and Replaced all *.feet / *.meters localization keys with generic ones (no “feet/meters” in the strings).

2) Let LanguageFormatter handle units:

So I Updated code to use LanguageFormatter.string(from:) so the inserted value is already like “10 meters” or “30 feet”. Updated localization placeholders to %@ since we’re inserting a String (not %d).

3) Fix threshold messaging:

So the “poor accuracy above …” message now formats the threshold (20m → “20 meters” / “60 feet”) using LanguageFormatter, so it stays consistent and never says something wrong.

Clean PR / whitespace:
I think this was done atleastt I tried to, but idk if it worked when I pushed it in 

Do tell if there is any other issues or errors you can find and ill fix as soon as I can.
Thanks for the feedback
-Ajitesh

JJGatchalian commented:
Is this pushing it a bit too far? Since 10 meters and 30 feet is not exactly the same, do you think it would be alright to hard code separate thresholds for imperial like 9 meters and 18 meters for the thresholds since they are a bit more accurate to the troubleshooting? I don't want to revise the troubleshooting saying about 33 feet is good, about 65 feet is ok...; it makes it harder for a beginner to relate.
